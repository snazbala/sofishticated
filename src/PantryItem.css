.pantry-item {
    border: 2px solid #FFFFFF;
    text-align: left;
    background: #FFFFFF;
    color: #364d4f;
    margin: 20px;
    padding: 20px;
    display: flex;
    justify-content: space-between;
}