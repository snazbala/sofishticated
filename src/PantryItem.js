import React, { Component } from 'react';
import './PantryItem.css';

class PantryItem extends Component {
    constructor(props) {
        super(props);
        this.state= {};
        this.removePantryItemHandler = this.removePantryItemHandler.bind(this);
    }

    removePantryItemHandler() {
        this.props.removePantryItem(this.props.name);
    }

    render() {
        return (
            <div className="pantry-item">
                {this.props.name} {this.props.quantity}
                &nbsp; &nbsp;
                <button className="remove" onClick={this.removePantryItemHandler}>Remove</button>
            </div>
        );
    }
};

export default PantryItem;