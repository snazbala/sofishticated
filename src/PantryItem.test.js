import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PantryItem from './PantryItem';

describe(PantryItem, () => {
    let name = 'Sugar';
    const mockRemovePantryItem = jest.fn();
    const component = shallow(
        <PantryItem name={name} removeItem={mockRemovePantryItem}/>
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <PantryItem name="Flour" />
        )
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains the supplied pantry item name', () => {
        expect(component.text()).toContain(name);
    });
});


