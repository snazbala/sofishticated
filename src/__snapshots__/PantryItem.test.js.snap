// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class PantryItem extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {};
        this.removePantryItemHandler = this.removePantryItemHandler.bind(this);
    }

    removePantryItemHandler() {
        this.props.removePantryItem(this.props.name);
    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'pantry-item', __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this },
                this.props.name, ' ', this.props.quantity, '\\xA0 \\xA0',

                _react2.default.createElement('button', { className: 'remove', onClick: this.removePantryItemHandler, __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this }, 'Remove')));


    }} renders and matches our snapshot 1`] = `
<div
  className="pantry-item"
>
  Flour
   
     
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
